clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitJobExecutor.java

```diff
@@ -75,6 +76,29 @@ public GitJobExecutor(
     askPassBinary = initAskPass();
   }
 
+  public void cloneOrPull(String repoUrl, String branch, Path localPath, String repoBasename)
+      throws IOException {
+    if (!Files.exists(localPath)
+        || localPath.toFile().listFiles() == null
+        || localPath.toFile().listFiles().length == 0) {
```

Comment:
Avoid calling `localPath.toFile().listFiles()` twice and might (unlikely) return different results.